pootle/core/views.py

```diff
@@ -509,6 +512,31 @@ def get_context_data(self, *args, **kwargs):
             'browse_url': self.browse_url}
 
 
+class PootleJSON(PootleDetailView):
+
+    response_class = JsonResponse
+
+    @never_cache
+    @set_permissions
+    @requires_permission("view")
+    def dispatch(self, request, *args, **kwargs):
+        if not settings.DEBUG and not request.is_ajax():
+            return HttpResponseBadRequest("This must be an AJAX request.")
+        return super(PootleDetailView, self).dispatch(request, *args, **kwargs)
+
+    def get_response_data(self, context):
+        """Override this method if you need to render a template with the context object
+        but wish to include the rendered output within a JSON response
+        """
+        return context
+
+    def render_to_response(self, context, **response_kwargs):
+        response_kwargs.setdefault('content_type', self.content_type)
```

Comment:
> And how do you pass it to self.render_edit_template()?

@unho this method is overridden to allow setting `response_data` separately from `context_data`